docs/developer/example-bpf.md

```diff
@@ -0,0 +1,670 @@
+```
+
+Then run with the privately built bytecode container image:
+
+```console
+    ./go-tc-counter -iface ens3 -direction ingress -location image://quay.io/$USER/go-tc-counter-bytecode:latest
+    2022/12/02 16:38:44 Reading /etc/bpfd/gocounter.toml ...
+    2022/12/02 16:38:44 Read /etc/bpfd/gocounter.toml failed: err open /etc/bpfd/gocounter.toml: no such file or directory
+    2022/12/02 16:38:44 Using Input: Interface=ens3 Priority=50 Source=quay.io/$USER/go-tc-counter-bytecode:latest
+    2022/12/02 16:38:45 Program registered with 0d313a4a-a17c-4c70-81ba-3ecc494b900e id
+    2022/12/02 16:38:48 4 packets received
+    2022/12/02 16:38:48 580 bytes received
+
+    2022/12/02 16:38:51 4 packets received
+    2022/12/02 16:38:51 580 bytes received
+
+    ^C2022/12/02 16:38:51 Exiting...
+    2022/12/02 16:38:51 Unloading Program: 0d313a4a-a17c-4c70-81ba-3ecc494b900e
+```
+
+#### Preloading BPF Bytecode
+
+Another way to load the BPF bytecode is to pre-load the BPF bytecode and
+pass the associated `bpfd` UUID to the userspace program.
+This is similar to how BPF programs will be loaded in Kubernetes, except `kubectl` commands will be
+used to create `bpfProgramConfig` CRD objects instead of using `bpfctl`, but that is covered in the next
+section.
+The userspace programs will skip the loading portion and use the UUID to find the shared
+map and continue from there.
+
+Referring back to the diagram above, the `load` and `unload` are being done by `bpfctl` and not
+`go-xdp-counter` userspace program.
+
+First, use `bpfctl` to load the `go-xdp-counter` BPF bytecode:
+
+```console
+    bpfctl load --location image://quay.io/bpfd-bytecode/go-xdp-counter:latest xdp --iface ens3 --priority 50
+    d541af30-69be-44cf-9397-407ee512547a
+```
+
+Then run the `go-xdp-counter` userspace program, passing in the UUID:
+
+```console
+    ./go-xdp-counter -iface ens3 -uuid d541af30-69be-44cf-9397-407ee512547a
+    2022/12/02 17:01:38 Using Input: Interface=ens3 Source=d541af30-69be-44cf-9397-407ee512547a
+    2022/12/02 17:01:41 180 packets received
+    2022/12/02 17:01:41 26100 bytes received
+
+    2022/12/02 17:01:44 184 packets received
+    2022/12/02 17:01:44 26680 bytes received
+
+    ^C2022/12/02 17:01:46 Exiting...
+```
+
+Then use `bpfctl` to unload the BPF bytecode:
+
+```console
+    bpfctl unload d541af30-69be-44cf-9397-407ee512547a
+```
+
+## Deploying On Kubernetes
+
+This section will describe loading bytecode on a Kubernetes cluster and launching the userspace
+program.
+The approach is slightly different when running on a Kubernetes cluster.
+The BPF bytecode should be loaded by an administrator, not the userspace program itself.
+
+![gocounter On Kubernetes](../img/gocounter-on-k8s.png)
+
+### Loading BPF Bytecode On Kubernetes
+
+This assumes there is already a Kubernetes cluster running and `bpfd` is running in the cluster
+(see [How to Manually Deploy bpfd on Kubernetes](../admin/k8s-deployment.md)).
```

Comment:
This like points to a super stake doc currently.... (I'm assuming what @anfredette Is doing will refactor it?)

Reply:
Yes, that's my assumption.